chall09/dalba-de.c

```diff
@@ -0,0 +1,189 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+typedef struct          	s_time
+{
+  	unsigned long long		year;
+	unsigned long long		day;
+	unsigned long long		hour;
+	unsigned long long		minute;
+	unsigned long long 		second;
+	char					*str;
+	char					*cyear;
+	char					*cday;
+	char					*chour;
+	char					*cminute;
+	char					*csecond;
+}                       	t_time;
+
+int					check_line(char *str)
+{
+	int i;
+
+	i = 0;
+  	if (str[i] == '\0')
+	return (-1);
+	while (str[i])
+	{
+		if (str[i] == ' ' || (str[i] >= '0' && str[i] <= '9'))
+			i++;
+		else
+			return (-1);
+	}
+	return (0);
+}
+
+unsigned long long	ft_atoi(char *line, int *i)
+{
+	unsigned long long	num;
+
+	num = 0;
+	while (line[*i] >= '0' && line[*i] <= '9')
+	{
+		num = num * 10 + (line[*i] - 48);
+		(*i)++;
+	}
+	return (num);
+}
+
+void        fill_second(t_time *t)
+{
+	char *gap;
+
+	gap = "";
+	if (t->year > 0 || t->day > 0 || t->hour > 0 ||t->minute > 0)
+		gap = " and ";
+	else
+		gap = "";
+  	t->csecond = malloc(sizeof(t->second) + 9 * sizeof(char));
+	if (t->second == 0)
+		sprintf(t->csecond, "");
+	else if (t->second == 1)
+		sprintf(t->csecond, "%s%llu second", gap, t->second);
+	else
+		sprintf(t->csecond, "%s%llu seconds", gap, t->second);
+}
+
+void      fill_minute(t_time *t)
+{
+	char *gap;
+
+	gap = "";
+	if ((t->year > 0 || t->day > 0 || t->hour > 0) && t->second == 0)
+		gap = " and ";
+	else if ((t->year > 0 || t->day > 0 || t->hour > 0) && t->second > 0)
+		gap = " , ";
+	else if (t->year == 0 && t->hour == 0 && t->minute == 0)
+		gap = "";
+  	t->cminute = malloc(sizeof(t->minute) + 8 * sizeof(char));
+	if (t->minute == 0)
+		sprintf(t->cminute, "");
+	else if (t->minute == 1)
+		sprintf(t->cminute, "%s%llu minute", gap, t->minute);
+	else
+		sprintf(t->cminute, "%s%llu minutes", gap, t->minute);
+}
+
+void      fill_hour(t_time *t)
+{
+	char *gap;
+
+	gap = "";
+	if ((t->year > 0 || t->day > 0) && (t->second == 0 && t->minute == 0))
+		gap = " and ";
+	else if ((t->year > 0 || t->day > 0) && (t->second > 0 || t->minute > 0))
+		gap = " , ";
```

Comment:
There should not be any space before de comma
```suggestion
		gap = ", ";
```